utils.js

```diff
@@ -0,0 +1,13 @@
+const getRepoFromProcess = () => process.env['INPUT_REPO'] ?? process.env['GITHUB_REPOSITORY']
+const createGetReleasesPath = (fullyQualifiedRepo = getRepoFromProcess()) => `/repos/${fullyQualifiedRepo}/releases`
+
+// The fact that this uses the "createGetReleasesPath" is used in a test, so changes here should be reflected there as
+// well.
+const createDeleteReleasesPath = (releaseId, fullyQualifiedRepo = getRepoFromProcess()) => `${createGetReleasesPath(fullyQualifiedRepo)}/${releaseId}`
```

Comment:
The idea of `fullyQualifiedRepo` is that it's the owner plus the name of a repo, since that always uniquely identifies a GitHub repo. I liked that idea so it's used pretty extensively by the tests but to keep the changes as small as possible, I didn't change the actual Action code to use that same pattern, but we definitely could. `owner` and `repo` in `lib.js` are never used independently; only together. Makes sense to just combine them once and never bother with it again.

That being said, it should always be explicitly provided by `lib.js`; `lib.js` relying on `getRepoFromProcess` would be bad since the `process.env` can be changed during the Action's run. We don't care in tests, though.

Reply:
I think we can go ahead and change the Action code.  I'd likely go for good testing (as you have done), and then we can do a prerelease of the action itself which signifies to community that it's not production ready (and so they can test and also give feedback).

The other thing is that I want to bump to eventually (maybe with all these changes) a major version like `v2.x` instead of the current `v0.2.1` since the Action has indeed been used by thousands of repos now from our insights and the semver would be easier understood with a real number against MAJOR.MINOR.PATCH 

Thoughts?

Reply:
Yeah, I think bumping would be good. But not to `v2.x`, but `v1.x`. Bumping to `2.x.x` isn't actually semver compatible; as far as I remember, all bumps have to be incremental by 1. So would have to bump to `v1.0.0`.

Reply:
It's semver compatible since semantic versioning doesn't say major version incremental by 1.  It says each element must increase numerically. And for major version, if any backwards incompatible changes are introduced, it must be incremented; it doesn't say HOW much ;-).  This has been brought up in the past, and covered in some of the FAQ Q/A like this one and a few others there:  https://semver.org/#what-if-i-inadvertently-alter-the-public-api-in-a-way-that-is-not-compliant-with-the-version-number-change-ie-the-code-incorrectly-introduces-a-major-breaking-change-in-a-patch-release

Reply:
but maybe "by 1" would indeed be less confusing for the uninitiated, so we'll do that way when the time comes; agree.